Responsive design with VUE

cagomezr opened the pull request:
Good day !

I enabled  responsive design and did some stylistic  choices on your program(mostly organization on mobile range.

I added a quick fix  that will allow your program to not throw  scroll around  as you rebuild  the "flex tables " I'm not sure how  Vue.JS does it, entirely (I have read superficially for  this project and a  few interviews)  but I think you are rebuilding tables more than once every time you do a  click event and henceforth throwing the scroll as it is rebuilding the flex tables again and moving scroll to last element at bottom to table. I added scroll to top to try to fix the issue, but I realize this may not be  the most elegant  solution to this problem,-nor correct one but didnt want to change your code without asking first or more research into problem

If you have any questions or feedback I can help you with this  further #1

Commit:
Update index.html

Adding respoinsive tags and   style issues

Commit:
Update index.html

Font changes

krzysztofrewak commented on index.html:
```diff
@@ -1,46 +1,55 @@
-				<div class="w-16 h-16 p-2 text-xs m-1 cursor-pointer flex justify-center text-center items-center uppercase" v-for="trait in traits" @click="setDisplayBy(trait)" :class="'bg-'+trait.background+'-400'" v-if="trait.groupable">
-					<span class="opacity-50">{{ trait.label }}</span>
+				<div class="w-16  h-16 p-2 text-xs m-1 cursor-pointer flex justify-start  lg:justify-center xl:justify-center text-center items-center uppercase" v-for="trait in traits" @click="setDisplayBy(trait)" :class="'bg-'+trait.background+'-400'" v-if="trait.groupable">
+					<span style="width:100%;" class="opacity-50 text-center">{{ trait.label }}</span>
```
You can use `w-full` class instead of `width: 100%` style property.

krzysztofrewak commented on index.html:
```diff
@@ -1,46 +1,55 @@
 		<link href="https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css" rel="stylesheet">
 		<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/components/icon.min.css">
 		<script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
+		<style>
```
Please remove unnecessary style tags from head.

krzysztofrewak commented on index.html:
```diff
@@ -121,17 +130,16 @@ <h2 class="uppercase text-gray-700 text-sm">commercial examples</h2>
 					if(this.tagFilter) {
 						styles = styles.filter(style => this.tagFilter.styles.includes(style.label.index))
 					}
+                    console.log("possible jump");
```
Please remove all console logs.

krzysztofrewak commented on index.html:
```diff
@@ -173,15 +181,13 @@ <h2 class="uppercase text-gray-700 text-sm">commercial examples</h2>
 				buildStyleLabel(style, index) {
 					let name = style.name.split(": ")
 					let id = style.id.split("-")
-
```
Why are you removing vertical density?

cagomezr replied:
Im not sure what you mean here  I compared the master  branch and my updated file and they are the same in this line Im confused.

cagomezr replied:
never mind after checking a  fresh  compare from github the  files are different , but doesnt show if you do a   quick copy paste?  well no matter its resolved it seems

krzysztofrewak commented on index.html:
```diff
@@ -195,25 +201,26 @@ <h2 class="uppercase text-gray-700 text-sm">commercial examples</h2>
 					if(this.selected) {
-						if(this.selected.id === style.id) {
+						if(this.selected.id === style.id) {  
+                             window.scrollTo(0,0); 
```
Why are you jumping to top three times in this method?

cagomezr replied:
I did this because there is a scroll issue on mobile  when you trigger the rebuild of  grid,  a single one didn't  solve the problem.  I believe its the machine  rebuilding page and  trying to go to end of page? - or at least looks that way I think  it may be a  quirk of  remembering scroll feature, but that seems unlikely, also it doesn't  trigger in all instances of  grid pressing.

krzysztofrewak commented on index.html:
```diff
@@ -1,46 +1,55 @@
-			<div class="flex p-8 flex-wrap justify-center bg-gray-100">
-				<div class="w-16 h-16 p-2 text-xs m-1 flex justify-center text-center items-center uppercase">
+			<div class="flex p-8 flex-wrap justify-start  lg:justify-center xl:justify-center bg-gray-100">
+			    <div class="flex p-4 flex-wrap">
```
I was using tabs in whole file, please respect that.

krzysztofrewak reviewed (changes requested):
Thank you for your contribution. I've highlighted some minor things I'd like you to fix to maintain code quality. Thanks again. :)

Commit:
Update index.html

Did  the asked  revisions
 Only issue is  I dont understand what you mean by vertical density?  I checked the file  and compared with master  the only time I affect it is when  changing your margins a bit on line 44 and 49 everything looks the same as in master branch

as for the  3 scroll jumps I removed them, but it was an strange quirk of the fix I did for your mobile jump issue I mentioned  one scroll didnt  seem to fix it but 3 seemed to do the trick I think is  still  the  page builder executing , and moving scroll all the way to end.

if you find any issues  let me know

cagomezr commented:
Hey I sent you some comments, the only issue I have is about the vertical density   Since I don't see it on comparison files both on notepad ++ and  the  comparison file on my end I can send you screenshots

Commit:
Update index.html

vertical density changes found

cagomezr commented:
I corrected all issues Hope this is  for your liking  let me know of any issues you may find

krzysztofrewak commented on index.html:
```diff
@@ -195,6 +201,7 @@ <h2 class="uppercase text-gray-700 text-sm">commercial examples</h2>
 							}
 						})
 					}
+					window.scrollTo(0,0);					 
```
Methods `buildStyleLabel()` and `styleBox()` are fired for every tile (with `styleBox()` fired with every tile action), so basically you're scrolling to top about... 12240 times. ;)

If you want to scroll page, you can add `window.scrollTo(0,0)` at the end of `selectStyle()` method.

cagomezr replied:
Moved the method, any idea why your scroll would jump though? after select Style has finished?  it doesnt jump every time ,  hit every time, which is what confuses me. the only time i been able to concistently figure out one  tag will make the scroll jump all the way down is if I press "Kellerbier" with all the others i get proper scrolling. maybe  something when you build or organize im not sure

Commit:
Update index.html

krzysztofrewak commented:
@cagomezr, thanks for your help! I merged your changes into master, it should be visible on https://krzysztofrewak.github.io/bjcp-ui/ soon. :)
